### PR TITLE
Wrap Results

### PR DESCRIPTION
**Pull request by khepin**
Results are now wrapped in a `Result` class. The result class tracks:

- whether the cache call was a hit or a miss
- the list of tags that are applied to the cached value

The result of calling `->load()` has been moved to a class called
`LoadResult` to clarify things since those 2 are fairly distinct.

### CODE REVIEW

**Comment by khepin** [comment]
Merging this would make a `2.x` version of the library. The current state of `master` has been tracked on a branch called `1-x` that has been pushed to this repo.

File: src/TTCache.php [hunk 1]
```diff
@@ -81,14 +80,14 @@ public function remember(string $key, ?int $ttl, array $tags, callable $cb)
...
-            return $r->value;
+            return Result::fromTaggedValue($r, true);
         }
 
         $r = $this->cache->get($hkey);
         if ($r) {
             $this->tree->child($r->tags);
             $this->resetTree($isRoot);
-            return $r->value;
+            return Result::fromTaggedValue($r, true);
```
Comment (bezhermoso) [inline src/TTCache.php:90]: Is there no tag info of value we can include here?

**Review by bezhermoso — COMMENTED** [review]
An upgrade guide would be helpful w/ these breaking changes.

File: src/TTCache.php (hunk 1 above)
Comment (khepin) [reply src/TTCache.php:90]: There is. `TaggedValue` is an internal class used for storing the data in MC. The `fromTaggedValue` factory method will grab the tags from that `TaggedValue` directly and apply them to the `Result`.

**Review by bezhermoso — APPROVED** [review]

**Pull request merged**